### PR TITLE
Ensure both LastModified and LastModifiedBy handle remove and removeAll

### PR DESCRIPTION
**Pull request by jlhughes**

### CODE REVIEW

File: src/foam/nanos/auth/LastModifiedAwareDAO.js [hunk 1]
```diff
@@ -32,6 +32,21 @@ foam.CLASS({
         }
         return super.put_(x, obj);
       `
+    },
+    {
+      name: 'remove_',
```
Comment (kgrgreer) [inline src/foam/nanos/auth/LastModifiedAwareDAO.js]: Isn't this the default behaviour anyway?

File: src/foam/nanos/auth/LastModifiedByAwareDAO.js [hunk 2]
```diff
@@ -28,11 +28,10 @@ foam.CLASS({
     },
     {
       name: 'remove_',
+      code: function(x, obj) {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/LastModifiedByAwareDAO.js]: Isn't this the default behaviour anyway?

File: src/foam/nanos/auth/LastModifiedAwareDAO.js [hunk 3]
```diff
@@ -32,6 +32,21 @@ foam.CLASS({
...
+        return this.SUPER(x, obj);
+      },
+      javaCode: `
+        return super.put_(x, obj);
+      `
+    },
+    {
+      name: 'removeAll_',
+      javaCode: `
+        getDelegate().select_(x, new foam.dao.RemoveSink(x, this), skip, limit, order, predicate);
```
Comment (kgrgreer) [inline src/foam/nanos/auth/LastModifiedAwareDAO.js]: What is this needed for?

**Comment by jlhughes** [comment]
Now 'consistent' the other way - less code.

**Pull request merged**